rmf_traffic/test/unit/agv/test_Negotiator.cpp

```diff
@@ -204,59 +205,113 @@ SCENARIO("Test Plan Negotiation Between Two Participants")
+        rmf_traffic::schedule::SimpleResponder::make(next_table));
 
-    CHECK(negotiation->ready());
+      CHECK(negotiation->ready());
 
-    next_table = negotiation->table(p2.id(), {p1.id()});
-    negotiator_2.respond(
-      next_table->viewer(),
-      rmf_traffic::schedule::SimpleResponder::make(next_table));
+      next_table = negotiation->table(p2.id(), {p1.id()});
+      negotiator_2.respond(
+        next_table->viewer(),
+        rmf_traffic::schedule::SimpleResponder::make(next_table));
 
-    CHECK(negotiation->complete());
+      CHECK(negotiation->complete());
 
-    auto proposals = negotiation->evaluate(
-      rmf_traffic::schedule::QuickestFinishEvaluator())->proposal();
-    REQUIRE(proposals.size() == 2);
+      auto proposals = negotiation->evaluate(
+        rmf_traffic::schedule::QuickestFinishEvaluator())->proposal();
+      REQUIRE(proposals.size() == 2);
 
-    const auto& proposal_1 = proposals.at(0);
-    const auto& proposal_2 = proposals.at(1);
-    for (const auto& r1 : proposal_1.itinerary)
-    {
-      for (const auto& r2 : proposal_2.itinerary)
+      const auto& proposal_1 = proposals.at(0);
+      const auto& proposal_2 = proposals.at(1);
+      for (const auto& r1 : proposal_1.itinerary)
       {
-        CHECK_FALSE(rmf_traffic::DetectConflict::between(
-            profile, r1->trajectory(),
-            profile, r2->trajectory()));
+        for (const auto& r2 : proposal_2.itinerary)
+        {
+          CHECK_FALSE(rmf_traffic::DetectConflict::between(
+              profile, r1->trajectory(),
+              profile, r2->trajectory()));
+        }
       }
+
+      //print_proposal(*proposals);
     }
 
-//    print_proposal(*proposals);
+    GIVEN("Negotiator #2 is a StubbornNegotiator")
+    {
+      rmf_traffic::schedule::StubbornNegotiator negotiator_2{ p2 };
+
+      auto next_table = negotiation->table(p1.id(), {});
+      negotiator_1.respond(
+        next_table->viewer(),
+        rmf_traffic::schedule::SimpleResponder::make(next_table));
+
+      REQUIRE(negotiation->table(p2.id(), {p1.id()}));
+
+      next_table = negotiation->table(p2.id(), {});
+      negotiator_2.respond(
+        next_table->viewer(),
+        rmf_traffic::schedule::SimpleResponder::make(next_table));
+
+      REQUIRE(negotiation->table(p1.id(), {p2.id()}));
+
+      next_table = negotiation->table(p1.id(), {p2.id()});
+      negotiator_1.respond(
+        next_table->viewer(),
+        rmf_traffic::schedule::SimpleResponder::make(next_table));
+
+      CHECK(negotiation->ready());
+
+      next_table = negotiation->table(p2.id(), {p1.id()});
+      // there should be a rejection of the proposed itinerary
+      negotiator_2.respond(
+        next_table->viewer(),
+        rmf_traffic::schedule::SimpleResponder::make(next_table));
+
+      // at this point, the negotiation is incomplete
+      CHECK_FALSE(negotiation->complete());
+
+      next_table = negotiation->table(p1.id(), {});
+      negotiator_2.respond(
```

Comment:
This is still using `negotiator_2` when it needs to be calling `negotiator_1` here.

The idea here is that `negotiator_1` needs to re-respond after `negotiator_2`'s rejection, because now `negotiator_1` has more information that it can use when making its plan.

Reply:
alright, that fixed it. thanks! 03b955f